arelle/ModelXbrl.py

```diff
@@ -22,35 +22,34 @@
 
 if TYPE_CHECKING:
     from arelle.ModelRelationshipSet import ModelRelationshipSet as ModelRelationshipSetClass
-    from arelle.ModelInstanceObject import ModelContext
+    from arelle.ModelInstanceObject import ModelContext, ModelFact, ModelDimension, ModelUnit
+    from arelle.ModelValue import QName, DATETIME
+    from datetime import date, datetime
+    from arelle import ModelFormulaObject, ModelManager
+
 
 
 ModelFact = None
 
-profileStatNumber = 0
+profileStatNumber: int = 0
 
-AUTO_LOCATE_ELEMENT = '771407c0-1d0c-11e1-be5e-028037ec0200' # singleton meaning choose best location for new element
-DEFAULT = sys.intern(str("default"))
-NONDEFAULT = sys.intern(str("non-default"))
-DEFAULTorNONDEFAULT = sys.intern(str("default-or-non-default"))
-EMPTY_TUPLE = ()
+AUTO_LOCATE_ELEMENT: str = '771407c0-1d0c-11e1-be5e-028037ec0200' # singleton meaning choose best location for new element
+DEFAULT: str = sys.intern(str("default"))
+NONDEFAULT: str = sys.intern(str("non-default"))
+DEFAULTorNONDEFAULT: str = sys.intern(str("default-or-non-default"))
+EMPTY_TUPLE: Tuple[()]
```

Comment:
This is a breaking change. `EMPTY_TUPLE` is no longer a defined value (also `typing.Tuple[x]` is deprecated, you should use `tuple[x]` instead).